### PR TITLE
Exclusion tables: In TTL

### PR DESCRIPTION
**Pull request by joeflack4**
Updates
```
    Exclusion tables
    - Update: Documentation. Put in appropriate README.md's in corresponding folders.
    - Update: Part of make command is now more explicit, e.g. replaced $(word N, $^) with path.
    - Update: Made CLI args required
    - Delete: CLI arg: onto-name
    - Add: CLI args: paths to each output file
    - Update: Output directory changed from config/ to reports/
    - Delete: Moved all outputs from config/ to reports/. This ended up in a delete, because of .gitignore. I believe this is intentional.
    - Misc: (i) makefile: robot -> $(ROBOT), (ii) makefile: Utilized iteration over ontology names.
    - Added: Required outpath param
    - Delete: Function to fill in missing CLI params.

    Misc
    - Update: Added exception for README in .gitignore
```

```
    - Added: context.json
```

```
    Exclusion tables
    - Added make targets for creating exclusion reasons in OBO and TTL, by ontology.
    - Added some TTL files generated from the above.
```

### CODE REVIEW

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:196]: @matentzn 
### Success cases
DOID, OMIM, ICD10WHO

### Failures
#### ORDO
```
sh run.sh make MIR=false config/ordo_exclusion_reasons.ttl
robot template --input mirror/ordo.owl --template config/ordo_exclusion_reasons.robot.template.tsv --output config/ordo_exclusion_reasons.ttl
UNKNOWN ENTITY ERROR could not interpret 'Orphanet:158048' in row 3, column 1 ("ID") in table "config/ordo_exclusion_reasons.robot.template.tsv".
```

#### ICD10CM
```
sh run.sh make MIR=false config/icd10cm_exclusion_reasons.ttl
robot template --input mirror/icd10cm.owl --template config/icd10cm_exclusion_reasons.robot.template.tsv --output config/icd10cm_exclusion_reasons.ttl
UNKNOWN ENTITY ERROR could not interpret 'ICD10CM:P90-P96' in row 3, column 1 ("ID") in table "config/icd10cm_exclusion_reasons.robot.template.tsv".
```

#### NCIT
```
sh run.sh make MIR=false config/ncit_exclusion_reasons.ttl
robot template --input mirror/ncit.owl --template config/ncit_exclusion_reasons.robot.template.tsv --output config/ncit_exclusion_reasons.ttl
ANNOTATION PROPERTY ERROR could not handle annotation property: MONDOREL:has_exclusion_reason
```

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:200]: @matentzn 
I ran `sh run.sh make config/doid_exclusion_reasons.obo`, but got this error:

> `Optional.get() cannot be called on an absent value`

I ran with `-vvv`. Here's the last, relevant part of the very long stack trace:
[optional get err.txt](https://github.com/monarch-initiative/mondo-ingest/files/9294640/optional.get.err.txt)

I ran for a couple ontologies and got the same issue, so I assume this will happen for any ontologies I try to run on this command, but not sure.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:193]: @matentzn Do you want (1) simply the OBO and TTL exclusion reasons per ontology? Or do you also want (2) A merged `.ttl` and `.obo` file as well? 

If you also want (2), how to do that? Pass multiple ontology `.owl` files to `--input`, and for `--template` I use the single merged `config/exclusion_reasons.robot.template.tsv`?

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:196]: They should all work now. Re-run.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:200]: Not doing OBO for now.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:193]: Nico wrote a merge command.

Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:193]: remove OBO from list of things. all good.

**Review by matentzn — APPROVED** [review]

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:190]: @matentzn I believe you said you wanted the output directory changed for all of these files from `config/` to `reports/` (except for the original exclusions files created by curators; I left those in `config/`).

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:198]: Combines the `.ttl` files. You wrote this code. I refactored to less lines.

File: src/ontology/config/README.md
Comment (joeflack4) [inline src/ontology/config/README.md:9]: Updated the `config/README.md`

File: src/ontology/reports/README.md
Comment (joeflack4) [inline src/ontology/reports/README.md:1]: Added `reports/README.md`. I also have this in another PR. I'll handle the merge conflict though when I rebase after whichever one is merged first.

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [inline src/scripts/exclusion_term_expansion.py:366]: @matentzn Added explicit `--outpath` param. I'm in the middle of checking if it works right now. My primary concern here is making sure that this works in the situation where we have a `make` target that has two files in the target, e.g. `$(REPORTDIR)/%_term_exclusions.txt $(REPORTDIR)/%_exclusion_reasons.robot.template.tsv:`.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:183]: Made each arg more explicit. TBH, this was handy for me to do for development, because then if I wanted to, I could comment out the prerequisites and run it faster. Otherwise, It got stuck on missing `$(word N,$^)

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:181]: @matentzn I found out why this was producing 0 results. It is because the [omim_exclusions.tsv.zip](https://github.com/monarch-initiative/mondo-ingest/files/9312932/omim_exclusions.tsv.zip) has empty columns for `exclude_children` and `exclusion_reason`.

If you want, I can (a) throw an error if something like this happens, or (b) do some default behavior.

@sabrinatoro @nicolevasilevsky Is there a single Google Drive folder that has all of the exclusion table Google Sheets? If so, can you give me permission and link me? If you guys have finished this for OMIM, I can download the results and run this again. Otherwise, I think we will probably need this filled out some time soon.

**Comment by joeflack4** [comment]
@matentzn You asked me to merge this when done, but I ended up changing more than expected, so I'd like you to take a final look and mark these comments resolved.

File: src/ontology/mondo-ingest.Makefile
Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:181]: I do the same thing all the time as well.

File: src/ontology/reports/README.md
Comment (matentzn) [inline src/ontology/reports/README.md:14]: (3)? Wrong numbering? Should be (4) right?

File: src/ontology/reports/README.md
Comment (matentzn) [inline src/ontology/reports/README.md:21]: (3)?

File: src/ontology/reports/component_signature-ncit.tsv
Comment (matentzn) [inline src/ontology/reports/component_signature-ncit.tsv:352]: Added this issue here: https://github.com/monarch-initiative/mondo-ingest/issues/66

BUt no need to deal with this as part of this here PR!

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [reply src/scripts/exclusion_term_expansion.py:336]: Agreed, all good. Thank you!

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [reply src/scripts/exclusion_term_expansion.py:362]: THANK YOU!

**Review by matentzn — APPROVED** [review]
I made a few comments. After addressing these as you see fit (or not), feel free to merge.

File: src/ontology/reports/README.md
Comment (joeflack4) [reply src/ontology/reports/README.md:14]: Ah yes. I moved these over to another file and renumbered them. I forgot that these numbers were referenced! Convenient but error-prone. I'll fix.

File: src/ontology/reports/README.md
Comment (joeflack4) [reply src/ontology/reports/README.md:21]: fixed! I changed to refer to the files by name rather than number, so this shouldn't happen again.

File: src/ontology/reports/component_signature-ncit.tsv
Comment (joeflack4) [reply src/ontology/reports/component_signature-ncit.tsv:352]: Sounds good. Responded to that issue with my thoughts. Will not address that in this PR.

**Pull request merged**
